src/main/scala/exercises/06_Collections.scala

```diff
@@ -129,34 +112,33 @@ object Collections {
    * Implement it whatever way suites you best. Hint: it can be done in a 
    * neat way using recursion. 
    */
-  def tails[T](l: List[T]): List[List[T]] = {
-    error("fix me")
+  def tails[T](l: List[T]): List[List[T]] = l match {
+    case List()  => List(l)
+    case _ :: xs => l :: tails(xs)
   }
   
   /**
    * Find the maximum element in a list, e.g. maxElementInList(List(1,9,3,5)) == 9
    * 
    * As usual, various ways exist: pattern matching, folding, ...
    */
-  def maxElementInList(l: List[Int]): Int = {
-    error("fix me")
-  }
+  def maxElementInList(l: List[Int]): Int = l.max
 
   /**
    * Calculate the sum of the equally position elements
    * of the two list
    */
-  def sumOfTwo(l1: List[Int], l2: List[Int]): List[Int] = {
-    error("fix me")
+  def sumOfTwo(l1: List[Int], l2: List[Int]): List[Int] = (l1, l2) match {
+    case (l, List())        => l
+    case (List(), l)        => l
+    case (x :: xs, y :: ys) => x + y :: sumOfTwo(xs, ys)
   }
 
   /**
    *  For this exercise preferably make use of the sumOfTwo
    * method above
    */
-  def sumOfMany(l: List[Int]*): List[Int] = {
-    error("fix me")
-  }
+  def sumOfMany(l: List[Int]*): List[Int] = l.foldLeft(List[Int]())(sumOfTwo)
```

Comment:
Using `List[Int]()` as a function argument results in a lot of parentheses which I think is a bit hard to read, using `List.empty[Int]` like Prof. Ben did in his solution might be nicer.